File: backtestApp/panels/base_panel.py
from PyQt5.QtWidgets import QGroupBox, QVBoxLayout
from abc import ABC, abstractmethod

class BasePanel(ABC):
    """Classe de base pour tous les panels de l'application."""
    
    def __init__(self, parent=None):
        self.parent = parent
        self.widgets = {}
    
    @abstractmethod
    def create(self) -> QGroupBox:
        """Méthode à implémenter par les classes dérivées pour créer le panel."""
        raise NotImplementedError("Les classes dérivées doivent implémenter cette méthode")
    
    def get_values(self):
        """Récupère les valeurs des widgets du panel sous forme de dictionnaire."""
        return {}
    
    def set_values(self, values):
        """Définit les valeurs des widgets du panel à partir d'un dictionnaire."""
        pass

File: cpp_backtestApp/src/panels/base_panel.cpp
#include "panels/base_panel.h"
#include <QDebug>

BasePanel::BasePanel(QWidget* parent)
    : m_parent(parent)
{
    // Initialisation simple du panel de base
}

BasePanel::~BasePanel()
{
    // Les widgets enfants seront détruits automatiquement par le parent
    // Ne pas supprimer m_parent car il est géré ailleurs
    qDebug() << "Destruction d'un panel";
}

QMap<QString, QVariant> BasePanel::getValues()
{
    // Implémentation par défaut : retourne une map vide
    // Les classes dérivées doivent surcharger cette méthode pour retourner les valeurs spécifiques
    qDebug() << "BasePanel::getValues() appelé - à surcharger dans les classes dérivées";
    return QMap<QString, QVariant>();
}

void BasePanel::setValues(const QMap<QString, QVariant>& values)
{
    // Implémentation par défaut : ne fait rien
    // Les classes dérivées doivent surcharger cette méthode pour définir les valeurs spécifiques
    Q_UNUSED(values);
    qDebug() << "BasePanel::setValues() appelé - à surcharger dans les classes dérivées";
}